File: scripts/upload_source_maps.py
import os
import sys
import json

num_args = len(sys.argv)

if num_args != 2:
	print("Usage: upload_source_maps.py <android|ios>")
	sys.exit(0)

target_platform = sys.argv[1]

with open("package.json", "r") as f:
	package = json.loads(f.read())

lines = [
	"curl -X POST --http1.1 https://upload.bugsnag.com/react-native-source-map",
	"-F apiKey=d6d84807bf87e7f452415373ff2070ed",
	f"-F appVersion={package['version']}",
	"-F dev=false",
	f"-F platform={target_platform}",
	"-F sourceMap=@android/app/build/generated/sourcemaps/react/release/index.android.bundle.map",
	"-F bundle=@android/app/build/generated/assets/react/release/index.android.bundle",
]

result = os.system(" ".join(lines))

print(result)

File: scripts/uploadSourceMaps.js
const pjson = require('../package.json');

const path = require("path");
const { upload } = require('bugsnag-sourcemaps');

upload({
    apiKey: 'd6d84807bf87e7f452415373ff2070ed',
    appVersion: pjson.version, // optional
    // codeBundleId: '1.0-123', // optional (react-native only)
    minifiedUrl: path.resolve(__dirname, '../android/app/build/generated/assets/react/release/index.android.bundle'), // supports wildcards
    sourceMap: path.resolve(__dirname, "../android/app/build/generated/sourcemaps/react/release/index.android.bundle.map"),
    minifiedFile: path.resolve(__dirname, '../android/app/build/generated/assets/react/release/index.android.bundle'), // optional
    overwrite: true, // optional
    // sources: {
    //     'http://example.com/assets/main.js': path.resolve(__dirname, 'path/to/main.js'),
    //     'http://example.com/assets/utils.js': path.resolve(__dirname, 'path/to/utils.js'),
    // },
}, function(err) {
    if (err) {
        throw new Error('Something went wrong! ' + err);
    }
    console.log('Sourcemap was uploaded successfully.');
});
